pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -251,8 +251,8 @@ func getExistingPodSpreadCache(pod *v1.Pod, nodeInfoMap map[string]*schedulernod
 	}
 	addTopologyPairMatchNum := func(pair topologyPair, num int32) {
 		lock.Lock()
+		defer lock.Unlock()
```

Comment:
please revert this line

Reply:
Emm... Can you elaborate on the reasons?
We already did like this https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/algorithm/predicates/metadata.go#L628
Using defer is more conventional!

Reply:
> Emm... Can you elaborate on the reasons?

defer adds extra cost, and we should avoid it. I don't think it worth it to use defer instead of call Unlock directly here.

FYI: https://github.com/golang/go/issues/14939

Reply:
+1 on @draveness. We should be only using `defer` when needed, e.g., multiple returns in a block.

Reply:
That makes sense! Thank you @draveness @Huang-Wei